doc/rfcs/inactive_users_cleanup.md

```diff
@@ -0,0 +1,45 @@
+# Inactive user cleanup
+When user registration finishes and users not passing "activation", corresponding DB record stays for ttl.
+But all metadata/alias/email records stay forever.
+
+1. Add additional Redis HASH and ZSET handling inactive users list.
+2. Change registration action:
+    - remove ttl assignment from from `USER_DATA` key
+    - call inactive list append process `addInactiveUser` 
+3. Change activation action:
+    - add inactive list cleanup `deleteInactiveUser`
+4. Add interval task.
+5. Add config parameter
+6. Move 'delete user' transaction function into separate file from `actions/remove.js` to `utils/removeUser.js`
+
+## Config changes
+Added new `deleteInactiveAccountsInterval` seconds to run cleaner task.
+
+## Constants
+New constants `USERS_ACTIVATE` - ZSET handling userId regtime, `USERS_ACTIVATE_AUDIENCE` - userId to audience binding
+
+## utils/inactiveUsers.js
+Handles all logic, including task creator, and database actions for InactiveUsersList
+
+### addInactiveUser(id, audience)
+Adds corresponding id record into `USERS_ACTIVATE` with score eq current time. Sets `USERS_ACTIVATE_AUDIENCE` hash to track audieces provided in registration process
+
+### deleteInactiveUser(id)
+Removes id dependent data from `USERS_ACTIVATE`, `USERS_ACTIVATE_AUDIENCE`
+
+### cleanInactiveUsers()
+Gets outdated users from `USERS_ACTIVATE` and deletes all user data from database(same as `@ms-users-api.remove`)
+
+
+#### Possible variant
+- Create additional set with id's of inactive users (`users-activate`).
+  * Change activation process: remove successfully activated user id from `users-activate`.
+  * Change registration process: insert inactive user id into `users-activate`.
+- Add repeating task with interval `config.incativeCleanupInterval`.
```

Comment:
I'd like to explore another venue where we perform cleanup for all inactive users during each registration and/or activation - process needs to very quick and written in LUA so that to be atomic - then we dont need to scan database and sets of pointers will be quick to use

Reply:
Please see https://github.com/makeomatic/ms-users/commit/9e25f85f2e41571ef990aa0c42440216d40e3f9c all audiences tracked on registration. When deleting user l, we clean all audiences. That's why I've moved remove user method outside action and use it for cleanup after some additional changes.

Reply:
Please see https://github.com/makeomatic/ms-users/commit/9e25f85f2e41571ef990aa0c42440216d40e3f9c all audiences tracked on registration. When deleting user l, we clean all audiences. That's why I've moved remove user method outside action and use it for cleanup after some additional changes.